src/Main/Views/Expenses/Expenses.js

```diff
@@ -1,46 +1,55 @@
-import React from 'react';
+import React, { useState } from 'react';
+
+import Table from "./Table/Table";
+import InputRow from "./InputRow/InputRow";
+import { inputOptions } from "./data/inputOptions";
+import { expensesData } from "./data/expensesData";
 
 import './Expenses.css';
 
-const expenses = () => {
+const Expenses = () => {
+
+    expensesData.forEach(entry => {
+        return entry.date = entry.date.slice(0, 10);
+    });
+
+    const [data, setData] = useState(expensesData);
+
+    const addEntries = () => {
+        const inputs = document.querySelectorAll(".Input");
+        const newObjKeys = ["date", "payee", "category", "memo", "outflow", "inflow"];
+        let inputsVal = [];
+        let newObj = {};
+
+        inputs.forEach(item => {
+            inputsVal.push(item.value)
+        });
+        newObjKeys.forEach((key, i) => {
+            newObj[key] = inputsVal[i];
+        });
+        if (newObj !== undefined) {
+            data.push(newObj);
+        }
+        setData(receivedData => [...receivedData]);
+    }
```

Comment:
Ogólnie przyznam, że wymyśliłaś bardzo ciekawy sposób tworzenia nowego entry :) 
Ale mam pytanie: 
1) setData() na końcu - co ono w zasadzie robi? => bo tak w zasadzie to wykorzystałaś dane wejściowe, by stworzyć kopie tablicy i ustawić ją jako data
2) * od razu taki tip: w useState() chodzi właśnie o to by nie operować bezpośrednio na data... data możesz użyć do odczytu albo utworzenia nowego obiektu, np. [...prevState, newValue] => i mam dziwne wrażenie, że o to chodziło w tym setData :)

Reply:
1. To jest BARDZO dobre pytanie. Po prostu nic innego mi nie działało, żeby dodawało się do tabelki. Nie miałam już innych pomysłów i w akcie desperacji po prostu spróbowałam xD 
2. Prawdopodobnie masz rację i nie wiem, czemu ten drugi argument z funkcją do update'owania stanu tak ciężko mi zakumać. 🤔

Reply:
@nettecode Poprawiłam to z setData nie operując właśnie już bezpośrednio na data, ale zastanawiam się, czy to normalne, że po zmianie zakładki, np. z Expenses na Settings i wtedy po powrocie do Expenses komponent jakby wraca do tego pierwszego zainicjowanego stanu... Wszystkie dodane wcześniej pozycje przepadają. Jak "utrzymać" stan między zmianami zakładek? 🤔

Reply:
Ten temat rozwinę w Notion, bo jest całkiem spory ;)